### PR TITLE
(MAINT) Update Puppet and other image components to latest versions

### PR DESCRIPTION
**Pull request by ahenroid**

### CODE REVIEW

**Comment by ahenroid** [comment]
I updated to the latest Puppet agent, server, etc. for use in demos, and it would be great to see this merged and the upstream images updated.

**Comment by garethr** [comment]
@ahenroid #25 has been merged. If you want to rebase we can merge this. I can then take a few folks through releasing.

**Comment by ahenroid** [comment]
I've rebased, but as I expected, the build failed. @garethr, I'm assuming you did the Travis setup, do you understand what is going on here?

File: facter/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM puppet/puppet-agent-ubuntu:1.6.2
+FROM puppet/puppet-agent-ubuntu:1.8.3
```
Comment (MWilsonPuppet) [inline facter/Dockerfile]: Change to puppet/puppet-agent-ubuntu:1.8.0

File: puppet-agent/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM puppet/puppet-agent-ubuntu:1.6.2
```
Comment (MWilsonPuppet) [inline puppet-agent/Dockerfile]: Change to puppet/puppet-agent-ubuntu:1.8.0

File: puppet-inventory/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM puppet/puppet-agent-ubuntu:1.6.2
+FROM puppet/puppet-agent-ubuntu:1.8.3
```
Comment (MWilsonPuppet) [inline puppet-inventory/Dockerfile]: Change to puppet/puppet-agent-ubuntu:1.8.0

File: puppetserver/Dockerfile [hunk 4]
```diff
@@ -1,7 +1,7 @@
-FROM puppet/puppetserver-standalone:2.6.0
+FROM puppet/puppetserver-standalone:2.7.2
```
Comment (MWilsonPuppet) [inline puppetserver/Dockerfile]: change to puppet/puppetserver-standalone:2.6.0

**Comment by MWilsonPuppet** [comment]
@ahenroid Hey Andy I took a look at what was going on and it appears that in any of the Docker files where we are using the images in Puppet repo on Dockerhub the repos have not been updated. I am going to speak to Gareth about how I go about updating these and will get them updated. I have checked the versions currently in use and added the comments to what they should be set and tested that this passed travis. This will get us the agent bump to 1.8.3 while I get the images on Dockerhub updated.

**Comment by ahenroid** [comment]
@MWilsonPuppet Ah! So we might have to stage these changes. That totally makes sense. Let me see if that works.

**Comment by MWilsonPuppet** [comment]
LGTM

**Comment by garethr** [comment]
Mmmm, I'm not sure about the comment in https://github.com/puppetlabs/puppet-in-docker/pull/26#issuecomment-275264092

I think this muddles the code and the images on Hub. The code is now out-of-sync with itself. ie. if you build all the images from the Dockerfiles, the dependent images don't use the latest local versions. I would probably have merged as is.

We have INFRA-126 open to do with this and updating the images on Hub. I can take folks through doing that.

**Comment by ahenroid** [comment]
@garethr, right you are. Sorry I've broken the the repo. Yes, whenever you have time, walk me, @MWilsonPuppet, and anyone else interested through the release process. Are there other PRs or issues that should be addressed/merged before the next release?

**Pull request merged**